PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:6 + add:0, change matches:
-    _inherit = ['account_edi_proxy_client.user']
-    _inherit = ['account.journal']
-    _inherit = ['account.move']
-    _inherit = ['res.company']
-    _inherit = ['res.config.settings']
-    _inherit = ['res.partner']

Total Changes: 14

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/account_peppol/models/account_edi_proxy_user.py
+++ b/addons/account_peppol/models/account_edi_proxy_user.py
@@ -13,7 +13,7 @@ BATCH_SIZE = 50
 
 
 class Account_Edi_Proxy_ClientUser(models.Model):
-    _inherit = ['account_edi_proxy_client.user']
+    _inherit = 'account_edi_proxy_client.user'
 
     peppol_verification_code = fields.Char(string='SMS verification code')
     proxy_type = fields.Selection(selection_add=[('peppol', 'PEPPOL')], ondelete={'peppol': 'cascade'})

--- a/addons/account_peppol/models/account_journal.py
+++ b/addons/account_peppol/models/account_journal.py
@@ -2,7 +2,7 @@ from odoo import _, fields, models
 
 
 class AccountJournal(models.Model):
-    _inherit = ['account.journal']
+    _inherit = 'account.journal'
 
     account_peppol_proxy_state = fields.Selection(related='company_id.account_peppol_proxy_state')
     is_peppol_journal = fields.Boolean(string="Account used for Peppol", default=False)

--- a/addons/account_peppol/models/account_move.py
+++ b/addons/account_peppol/models/account_move.py
@@ -5,7 +5,7 @@ from odoo.exceptions import UserError
 
 
 class AccountMove(models.Model):
-    _inherit = ['account.move']
+    _inherit = 'account.move'
 
     peppol_message_uuid = fields.Char(string='PEPPOL message ID')
     peppol_move_state = fields.Selection(

--- a/addons/account_peppol/models/account_move_send.py
+++ b/addons/account_peppol/models/account_move_send.py
@@ -5,7 +5,7 @@ from odoo.addons.account_edi_proxy_client.models.account_edi_proxy_user import A
 
 
 class AccountMoveSend(models.AbstractModel):
-    _inherit = ['account.move.send']
+    _inherit = 'account.move.send'
 
     # -------------------------------------------------------------------------
     # ALERTS

--- a/addons/account_peppol/models/res_company.py
+++ b/addons/account_peppol/models/res_company.py
@@ -47,7 +47,7 @@ PEPPOL_ENDPOINT_SANITIZERS = {
 
 
 class ResCompany(models.Model):
-    _inherit = ['res.company']
+    _inherit = 'res.company'
 
     account_peppol_contact_email = fields.Char(
         string='Primary contact email',

--- a/addons/account_peppol/models/res_config_settings.py
+++ b/addons/account_peppol/models/res_config_settings.py
@@ -7,7 +7,7 @@ from odoo.addons.account_peppol.tools.demo_utils import handle_demo
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     account_peppol_edi_user = fields.Many2one(
         comodel_name='account_edi_proxy_client.user',

--- a/addons/account_peppol/models/res_partner.py
+++ b/addons/account_peppol/models/res_partner.py
@@ -15,7 +15,7 @@ TIMEOUT = 10
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     invoice_sending_method = fields.Selection(
         selection_add=[('peppol', 'by Peppol')],
